cucumber-selenium-tests/src/test/java/com/skillstorm/PageObjects/TransactionPage.java

```diff
@@ -0,0 +1,281 @@
+package com.skillstorm.PageObjects;
+
+import java.time.Duration;
+import java.util.List;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import com.skillstorm.WebDriverSingleton;
+import com.skillstorm.PageObjects.Components.Navbar.DashboardNavbar;
+import com.skillstorm.PageObjects.Interfaces.Component;
+
+public class TransactionPage extends Page{
+
+    WebDriver driver = WebDriverSingleton.getDriver();
+
+    //static strings for IDs
+    private static final String editTransactionAmountFieldId = "edit-transaction-amount";
+    private static final String editTransactionNameField = "edit-transaction-vendorName";
+
+    // Locators for Transaction Page Elements
+    @FindBy(className = "usa-logo__text")
+    private WebElement transactionPageTitle;
+
+    @FindBy(id = "clearFilterBtn")
+    private WebElement clearFilterBtn;
+
+
+    @FindBy(id = "sortByDropdown")
+    private WebElement sortByDropdown;
+
+
+    @FindBy(id = "directionDropdown")
+    private WebElement directionDropdown;
+
+
+    @FindBy(id = "addTransactionModal")
+    private WebElement addTransactionModal;
+
+
+    // Modal Fields
+    @FindBy(name = "vendorName")
+    private WebElement vendorNameField;
+
+    @FindBy(name = "accountId")
+    private WebElement accountDropdown;
+
+    @FindBy(name = "amount")
+    private WebElement amountField;
+
+    @FindBy(name = "category")
+    private WebElement categoryDropdown;
+
+    @FindBy(id = "addTransactionBtn")
+    private WebElement submitBtn;
+
+
+    // Filter Dropdowns
+    @FindBy(id = "allCategoriesDropDown")
+    private WebElement allCategoriesDropDown;
```

Comment:
Low impact: Different naming conventions within the same file (see accountDropdown or categoryDropdown)